clients/src/test/java/org/apache/kafka/clients/admin/NewTopicTest.java

```diff
@@ -0,0 +1,168 @@
+        NewTopic newTopic = new NewTopic(TEST_TOPIC, replicasAssignments);
+        Map<Integer, List<Integer>> returnedAssignments = newTopic.replicasAssignments();
+
+        assertThrows(UnsupportedOperationException.class, () -> {
+            returnedAssignments.put(1, Arrays.asList(3, 4));
+        });
+    }
+
+    @Test
+    public void testConvertToCreatableTopicWithPartitionsAndReplicationFactor() {
+        NewTopic newTopic = new NewTopic(TEST_TOPIC, NUM_PARTITIONS, REPLICATION_FACTOR);
+        CreateTopicsRequestData.CreatableTopic creatableTopic = newTopic.convertToCreatableTopic();
+
+        assertEquals(TEST_TOPIC, creatableTopic.name());
+        assertTrue(creatableTopic.numPartitions() > 0);
+        assertEquals(NUM_PARTITIONS, creatableTopic.numPartitions());
+        assertTrue(creatableTopic.replicationFactor() > 0);
+        assertEquals(REPLICATION_FACTOR, creatableTopic.replicationFactor());
+    }
+
+    @Test
+    public void testConvertToCreatableTopicWithReplicasAssignments() {
+        int partitionIndex = 0;
+        Map<Integer, List<Integer>> replicasAssignments = new HashMap<>();
+        replicasAssignments.put(partitionIndex, BROKER_IDS);
+        NewTopic topic = new NewTopic(TEST_TOPIC, replicasAssignments);
+        Map<String, String> configs = new HashMap<>();
+        configs.put(CLEANUP_POLICY_CONFIG_KEY, CLEANUP_POLICY_CONFIG_VALUE);
+        topic.configs(configs);
+
+        CreateTopicsRequestData.CreatableTopic creatableTopic = topic.convertToCreatableTopic();
+
+        assertEquals(TEST_TOPIC, creatableTopic.name());
+        assertEquals(NO_NUM_PARTITIONS, creatableTopic.numPartitions());
+        assertEquals(NO_REPLICATION_FACTOR, creatableTopic.replicationFactor());
+        assertNotNull(creatableTopic.assignments());
+        assertEquals(1, creatableTopic.assignments().size());
+
+        CreateTopicsRequestData.CreatableReplicaAssignmentCollection assignmentsCollection = creatableTopic.assignments();
+
+        CreateTopicsRequestData.CreatableReplicaAssignment assignment = assignmentsCollection.find(partitionIndex);
+        assertEquals(partitionIndex, assignment.partitionIndex());
+        assertEquals(BROKER_IDS, assignment.brokerIds());
+
+        assertNotNull(creatableTopic.configs());
+        assertEquals(1, creatableTopic.configs().size());
+
+        CreateTopicsRequestData.CreateableTopicConfig config = creatableTopic.configs().find(CLEANUP_POLICY_CONFIG_KEY);
+        assertEquals(CLEANUP_POLICY_CONFIG_KEY, config.name());
+        assertEquals(CLEANUP_POLICY_CONFIG_VALUE, config.value());
+    }
+
+    @Test
+    public void testToString() {
+        NewTopic topic = new NewTopic(TEST_TOPIC, NUM_PARTITIONS, REPLICATION_FACTOR);
+        String expected = "(name=" + TEST_TOPIC + ", numPartitions=" + NUM_PARTITIONS + ", replicationFactor=" + REPLICATION_FACTOR + ", replicasAssignments=null, configs=null)";
+        assertEquals(expected, topic.toString());
```

Comment:
maybe we can verify the non-null configs/assignments too?

Reply:
Nice one, updated.